src/tasks/__tests__/processBulkUpload.int.test.ts

```diff
@@ -369,7 +378,31 @@ describe('processBulkUpload', () => {
 			getMockJobHelpers(),
 		);
 		const updatedBulkUpload = await loadBulkUpload(bulkUpload.id);
-		expect(updatedBulkUpload.status).toEqual(BulkUploadStatus.FAILED);
+		expect(updatedBulkUpload).toMatchObject({
+			status: BulkUploadStatus.FAILED,
+			fileSize: 93,
+		});
+	});
+
+	it('should update the file size for the bulk upload if the sourceKey is accessible and contains a valid CSV', async () => {
+		await createTestBaseFields();
+		const sourceKey = TEST_UNPROCESSED_SOURCE_KEY;
+		const bulkUpload = await createTestBulkUpload({ sourceKey });
+		await mockS3ResponsesForBulkUploadProcessing(
+			bulkUpload,
+			`${__dirname}/fixtures/processBulkUpload/validCsvTemplate.csv`,
+		);
+
+		await processBulkUpload(
+			{
+				bulkUploadId: bulkUpload.id,
+			},
+			getMockJobHelpers(),
+		);
+		const updatedBulkUpload = await loadBulkUpload(bulkUpload.id);
+		expect(updatedBulkUpload).toMatchObject({
+			fileSize: 93,
+		});
```

Comment:
I wonder if we should be testing that the file size actually changed from a prior value? In other words, are we sufficiently covered to know "after processing, it is populated with the expected value", or do we also want to know that it's the processing, specifically, which is populating it?

If so, we should verify that it's `null` before processing, or at least verify it changed.

What do you think?

Reply:
Gut feeling is that I generally think it's OK to not verify that our test setup was properly setup *so long as it's easy to see what the set up is by viewing the test*.  For instance for tests of `update` endpoints I wouldn't personally want an `expect(before).toBe({thethingwejustenetered})` since that feels redundant...

But reasonable people might disagree!

All that said, this is a weird case because we don't pass `null` in, rather, we don't pass *anything* in when creating the bulk upload, since you cannot create a bulk upload with a non-null value...

I'll sleep on it, but for now am inclined to think that the fact that it's impossible to insert a file size when creating a bulk upload is good enough reason to not verify starting state explicitly.

Reply:
> you cannot create a bulk upload with a non-null value

Oh! I didn't know this.

I'm fine going either way on this, follow your thinking.

That said, if you're trying to test that `loadBulkUpload` actually *does* something, then I do think the test needs to verify that it did something, which would mean monitoring its before and after state. Otherwise it could just be no-op'ing. Even though _you know_ it can't be, that relies on understanding the internals of the `create` setup. It feels a bit too implicit to me? But genuinely think I'm splitting hairs here, I'm comfortable either way.